src/main-client-v2.ts

```diff
@@ -3190,6 +3191,328 @@ export class MainClient extends BaseRestClient {
+  }
+
+  universalTransferBroker(
+    params: UniversalTransferBrokerParams,
+  ): Promise<BrokerSubAccount> {
+    return this.postPrivate('sapi/v1/broker/universalTransfer', params);
+  }
+
+  getUniversalTransferBroker(
+    params: GetUniversalTransferBrokerParams,
+  ): Promise<BrokerSubAccount> {
+    return this.getPrivate('sapi/v1/broker/universalTransfer', params);
+  }
+
+  /**
+   *
+   * EXCHANGE LINK - Fee Endpoints
+   *
+   */
+
+  // USD & Coin-M can be found under API getIncome() (find "API rebate" in results)
+  getBrokerSpotRebateHistory(days: 7 | 30, customerId?: string) {
+    if (days === 7) {
+      return this.getPrivate('sapi/v1/apiReferral/rebate/recentRecord', {
+        customerId,
+      });
+    }
+    if (days === 30) {
+      return this.getPrivate('sapi/v1/apiReferral/rebate/historicalRecord', {
+        customerId,
+      });
+    }
+  }
+
+  /**
+   * Broker Endpoints - only on old docs
+   */
+
+  /**
+   * Only in old docs
+   **/
+  getBrokerIfNewSpotUser(): Promise<{
+    rebateWorking: boolean;
+    ifNewUser: boolean;
+  }> {
+    return this.getPrivate('sapi/v1/apiReferral/ifNewUser');
+  }
+
+  /**
+   * Only in old docs
+   **/
+  getBrokerSubAccountDepositHistory(
+    params?: GetBrokerSubAccountDepositHistoryParams,
+  ): Promise<SubAccountDepositHistoryList[]> {
+    return this.getPrivate('sapi/v1/bv1/apiReferral/ifNewUser', params);
+  }
+
+  /**
+   * Only in old docs
+   **/
+  getBrokerUserCustomisedId(market: 'spot' | 'futures') {
+    const prefix = market === 'spot' ? 'sapi' : 'fapi';
+    return this.getPrivate(prefix + '/v1/apiReferral/userCustomization');
+  }
+
+  /**
+   * Only in old docs
+   **/
+  enableFuturesBrokerSubAccount(
+    params: EnableFuturesBrokerSubAccountParams,
+  ): Promise<EnableFuturesBrokerSubAccountResponse> {
+    return this.postPrivate('sapi/v1/broker/subAccount', params);
+  }
+
+  /**
+   * Only in old docs
+   **/
+  enableMarginApiKeyBrokerSubAccount(
+    params: EnableMarginApiKeyBrokerSubAccountParams,
+  ): Promise<BrokerSubAccount> {
+    return this.postPrivate('sapi/v1/broker/subAccount/margin', params);
+  }
+
+  /**
+   * Validate syntax meets requirements set by binance. Log warning if not.
+   */
+  private validateOrderId(
+    params:
+      | NewSpotOrderParams<any, any>
+      | CancelOrderParams
+      | NewOCOParams
+      | CancelOCOParams
+      | NewOrderListParams<any>,
+    orderIdProperty: OrderIdProperty,
+  ): void {
+    const apiCategory = 'spot';
+    if (!params[orderIdProperty]) {
+      params[orderIdProperty] = generateNewOrderId(apiCategory);
+      return;
+    }
+
+    const expectedOrderIdPrefix = `x-${getOrderIdPrefix(apiCategory)}`;
+    if (!params[orderIdProperty].startsWith(expectedOrderIdPrefix)) {
+      logInvalidOrderId(orderIdProperty, expectedOrderIdPrefix, params);
+    }
+  }
+
+  /**
+   *
+   * User Data Stream Endpoints
+   *
+   **/
```

Comment:
Here I added User Data stream, grouped as they were. Will also double check if we are missing any by going through all of them  after this is done :)